BadApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using BadApi.Controllers.Base;
+using BadApi.Domain;
+using BadApi.Models.Post;
+using Dapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace BadApi.Controllers
+{
+	[RoutePrefix("api/post")]
+    public class PostController : BaseApiController
+    {
+		private static List<ViewPostModel> _postCache = new List<ViewPostModel>();
+
+		[HttpGet]
+		[Route("")]
+		public async Task<ViewPostModel> GetPost(long id)
+		{
+			try
+			{
+				if (_postCache.Any(p => p.PostId == id))
+				{
+					return _postCache.First(p => p.PostId == id);
+				}
+
+				using (var connection = GetConnection())
+				{
+					var post = await connection.QueryFirstAsync<Post>(string.Format("SELECT * FROM dbo.Post WHERE PostId = {0}", id));
+					var myRank = await connection.QueryFirstOrDefaultAsync<Rank>(string.Format("SELECT * FROM dbo.Rank WHERE PostId = {0} AND AccountId = 1", id));
+					var positives = await connection.ExecuteScalarAsync<int>(string.Format("SELECT COUNT(*) FROM dbo.Rank WHERE PostId = {0} AND IsPositive = 1", id));
```

Comment:
Consider using named variables to represent True and False rather than hardcoding 1 and 0 in the query